raster/r.horizon/main.c

```diff
@@ -844,6 +844,8 @@ void calculate_shadow(void)
             shadow_angle *= rad2deg;
         }
         printangle = angle * rad2deg - 90.;
+        if (fabs(printangle) < GRASS_EPSILON)
+            printangle = fabs(printangle);
         if (printangle < 0.)
             printangle += 360;
```

Comment:
For adding and substracting 360 to make an angle be between 0-360, the angle before this needs to be in a certain range, or else more than one full circle would be needed.
I tried to follow the chain from where the input was: 
`printangle` --> `angle` --> `single_direction` --> `parm.direction->answer`
the parameter from the input is a double
https://github.com/OSGeo/grass/blob/76c1ac531f69c80f88de127fb38819123e9e4784/raster/r.horizon/main.c#L389-L390
https://github.com/OSGeo/grass/blob/76c1ac531f69c80f88de127fb38819123e9e4784/raster/r.horizon/main.c#L195-L201
But the range is not checked anywhere (that I'm aware of).

If you use a direction of -360*10-12=-3612.0 degrees, it won't work, but would accept my input.

Reply:
Would adding this help?

`parm.direction->options = "0-360";`